src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs

```diff
@@ -50,6 +54,48 @@ internal static class FixAllContextHelper
 
                     break;
 
+                case FixAllScope.ContainingMember or FixAllScope.ContainingType:
+                    if (document != null && !await document.IsGeneratedCodeAsync(cancellationToken).ConfigureAwait(false))
+                    {
+                        if (!fixAllSpans.IsEmpty)
+                        {
+                            // We are processing a span-based FixAllContext created for a partial declaration.
+                            var documentDiagnostics = ImmutableArray<Diagnostic>.Empty;
+                            foreach (var fixAllSpan in fixAllSpans)
+                            {
+                                documentDiagnostics = documentDiagnostics.AddRange(await fixAllContext.GetDocumentSpanDiagnosticsAsync(
+                                    document, fixAllSpan).ConfigureAwait(false));
+                            }
+
+                            return ImmutableDictionary<Document, ImmutableArray<Diagnostic>>.Empty.SetItem(document, documentDiagnostics);
+                        }
+                        else if (triggerSpan.HasValue)
+                        {
+                            // We need to compute diagnostics for each of the containing member/type and its partial declarations
+                            // using the trigger span.
+                            var documentsAndSpans = await GetDocumentsAndSpansForContainingMemberOrTypeAsync(document,
+                                fixAllContext.Scope, triggerSpan.Value, fixAllContext.CancellationToken).ConfigureAwait(false);
+                            var unused = PooledDictionary<Document, ImmutableArray<Diagnostic>>.GetInstance(out var builder);
```

Comment:
we have helpers already that go from an mutable dict builder with ArrayBuilder values, which can then be converted to an immutable/immutable collection.  The extension is called MultiAdd, and you can see the pattern in CodeFixService.cs i believe.

Reply:
None of the existing overloads work for this case as we need to append array of diagnostics in each loop iteration. Also given we don't expect multiple fix all spans for each document in majority of cases, we should be fine here.